browser/ui/views/frame/brave_browser_non_client_frame_view_mac.mm

```diff
@@ -96,5 +96,7 @@
 
   // In case title visibility wasn't changed and only vertical tab strip enabled
   // state changed, we should reset controls positions manually.
-  frame()->ResetWindowControlsPosition();
+  base::SequencedTaskRunnerHandle::Get()->PostTask(
+      FROM_HERE, base::BindOnce(&views::Widget::ResetWindowControlsPosition,
+                                base::Unretained(frame())));
```

Comment:
Q) Can we guarantee that `frame()` is always valid when `ResetWindowControlsPosition()`  is called?

Reply:
Yeah I think so because frame() is a widget and this view is a child of that widget.

Reply:
Well... it seems `ResetWindowControlsPosition()` could be called even this view is desroyed because bound callback's life cycle is not tied with this class?

Reply:
I thought ResetWindowControlsPosition() isn't this view's method and the widget would be alive on the next tick of the run loop. But that could be too naive. Changed to frame()->GetWeakPtr().